packages/core/lib/dlc/PolynomialPayoutCurve.ts

```diff
@@ -0,0 +1,187 @@
+    const { left, slope } = this;
+
+    const x = new BigNumber(Number(outcome));
+
+    // y = mx + b
+    const payout = slope.times(x.minus(left.outcome)).plus(left.payout);
+
+    return payout;
+  }
+
+  /**
+   * Get the outcome for a given payout
+   * @param payout The payout to get the outcome for
+   * @returns The outcome for the payout
+   */
+  getOutcomeForPayout(payout: BigNumber): bigint {
+    const { left, slope } = this;
+    const y = new BigNumber(Number(payout));
+
+    // Find the x value for the given y
+    // slope = (y2 - y1) / (x2 - x1)
+    // x1 = (y2 - y1) / slope + x2
+    const outcome = y
+      .minus(left.payout)
+      .dividedBy(slope)
+      .plus(left.outcome)
+      .integerValue();
+    return BigInt(outcome.toString());
+  }
+
+  /**
+   * Serializes PolynomialPayoutCurve to a PolynomialPayoutCurvePiece (for transport)
+   * @returns A PolynomialPayoutCurvePiece
+   */
+  toPayoutCurvePiece(): PolynomialPayoutCurvePiece {
+    const { points } = this;
+
+    const piece = new PolynomialPayoutCurvePiece();
+
+    piece.points = points.map((point) => {
+      const eventOutcome = BigInt(point.outcome.toString());
+      const outcomePayout = BigInt(point.payout.toString());
+      const extraPrecision = getPrecision(point.payout);
+      return { eventOutcome, outcomePayout, extraPrecision };
+    });
+
+    return piece;
+  }
+
+  /**
+   * Determine if the payout curve is equal to another
+   * @param curve A PolynomialPayoutCurve
+   * @returns True if the curves are the same
+   */
+  equals(curve: PolynomialPayoutCurve): boolean {
+    if (this.points.length !== curve.points.length) return false;
+
+    return this.points.every((point, i) => {
+      const otherPoint = curve.points[i];
+      return (
+        point.outcome.eq(otherPoint.outcome) &&
+        point.payout.eq(otherPoint.payout)
+      );
+    });
+  }
+
+  /**
+   * Creates a PolynomialPayoutCurve from a PolynomialPayoutCurvePiece
+   * @param piece
+   * @returns A PolynomialPayoutCurve
+   */
+  static fromPayoutCurvePiece(
+    piece: PolynomialPayoutCurvePiece,
+  ): PolynomialPayoutCurve {
+    const points = piece.points.map((point) => {
+      const outcome = new BigNumber(point.eventOutcome.toString());
+      const payout = new BigNumber(point.outcomePayout.toString()).plus(
+        fromPrecision(point.extraPrecision),
+      );
+      return { outcome, payout };
+    });
+    return new PolynomialPayoutCurve(points);
+  }
+
+  /**
+   * Computes all CETs for a given payout curve
+   * @param payoutFunction The payout function
+   * @param totalCollateral The total collateral
+   * @param roundingIntervals The rounding intervals
+   * @returns A list of CETs
+   */
+  static computePayouts(
+    payoutFunction: PayoutFunctionV0,
+    totalCollateral: bigint,
+    roundingIntervals: RoundingIntervalsV0,
+  ): CETPayout[] {
+    if (payoutFunction.pieces.length !== 3)
+      throw new Error('Must have at least three pieces');
```

Comment:
Shouldn't `payoutFunction.pieces.length` only be `2` if we're dealing with a line? 

And it would be 3 for Quadratic, and 4 for Cubic

Line: https://github.com/bitcoin-s/bitcoin-s/blob/master/core/src/main/scala/org/bitcoins/core/protocol/dlc/models/DLCPayoutCurve.scala#L421

Reply:
@matthewjablack 
<img width="626" alt="image" src="https://user-images.githubusercontent.com/12267041/158523272-d6efa111-0e12-476c-95fd-3004b492baef.png">

The three pieces are for the max loss, actual payout curve, and max gain (3 lines in total)

Reply:
A **line curve piece** consists of two points.
There are three line curve pieces that make up our payout function.

Reply:
Great question! Yeah as Steven mentioned, our `PayoutFunction` consists of three separate `PolynomialPayoutCurve`'s (in order to set a max loss and max gain), and each curve is made up of two points.

Reply:
Ah I see it's representing lines not points. Makes sense. Thanks for clarification.